docs/modules/reference/pages/resources.adoc

```diff
@@ -0,0 +1,137 @@
+= Resources
+:default_api_ns: io.pedestal.http.resources
+
+Although most of Pedestal is focused on _dynamic_ routes, nearly all HTTP-based applications will have some amount of static resources that need to be client accessible as well.
+
+These resources take the form of either plain files on the file system, or virtual files available on the https://docs.oracle.com/javase/tutorial/essential/environment/paths.html:[Java classpath].
+
+The classpath includes not just the code of your application, but the code for the Pedestal library, and any other dependencies you have defined for your project.
+
+For both classpath and file system resources, you must be careful about what you expose - only specific directories, known to be free of any compromising information (such as API keys or other secrets).
+
+== Resource Routes
+
+The preferred approach is to add routes for resources to your application.
+
+[source,clojure]
+----
+(ns org.example.service
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.resources :as resources]
+            [io.pedestal.http.route :as route]))
+
+(defn create-server
+  []
+  (http/create-server
+    {::http/port 8890
+     ::http/type :jetty
+     ::http/join? false
+     ::http/routes (route/routes-from
+                     #{ ... }  <1>
+                     (resources/file-routes {:file-root "web/public"}))})) <2>
+----
+
+<1> This is where you would put the dynamic routes for your application.
+<2> This exposes any files stored in the `web/public` folder under the default URI path `/public`.
+
+The `routes-from` macro will expand and combine the routes together to form the full routing table.
+
+The api:resource-routes[] function is analagous to `file-routes`, but returns routes that expose resources on the classpath.
+
+Each of these functions take a single parameter, a map of options:
+
+.Options to file and resource functions
+[options=header,cols="1,1,^1,2p"]
+|===
+| Option | Default | Only | Description
+
+| :prefix          | "/public" |          | The path prefix for the exposed resources.
+| :root-path       | -         | file     | Path to root directory to expose.
+| :resource-root   | -         | resource | Prefix on resources, limiting access to a safe subset of the classpath
+| :classloader     | -         | resource | Classloader used to find resources, or nil for the default class loader
+| :index-files?    | true      | file     | If true, paths that map to directories may return the index file for that directory.
+| :cache?          | true      |          | If true, then resource data is cached to in memory to speed up requests.
```

Comment:
This should probably default to false since the in-memory cache is unbounded (and hot resources are likely to be loaded from the OS page cache anyway)

Reply:
I'd prefer to leave this defaulted to true, as that's sensible for most applications that may have dozens, or even hundreds of resources to be exposed.  An application that have thousands of resources, or resources that might change during execution, should disable the cache, and perhaps, implement their own caching strategy as an interceptor around the routes.